homeassistant/components/mqtt/select.py

```diff
@@ -148,27 +158,26 @@ def _setup_from_config(self, config):
             ).async_render_with_possible_json_value,
         }
 
-    def _prepare_subscribe_topics(self):
+    def _prepare_subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
 
         @callback
         @log_messages(self.hass, self.entity_id)
-        def message_received(msg):
+        def message_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT messages."""
+            payload: ReceivePayloadType | None
             payload = self._templates[CONF_VALUE_TEMPLATE](msg.payload)
-
-            if payload.lower() == "none":
+            if payload and payload.lower() == "none":
                 payload = None
 
-            if payload is not None and payload not in self.options:
+            if isinstance(payload, bytes) or payload and payload not in self.options:
```

Comment:
I think it's more efficient to have the `None` check first. `isinstance` can be expensive so if we can return earlier it's better.
```suggestion
            if payload is not None and (isinstance(payload, bytes) or payload not in self.options):
```

Reply:
payload will never be None, since the template returns `str` | `bytes`. We can do a str(payload) here to solve this.